sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientTest.java

```diff
@@ -210,6 +210,42 @@ void receivesNumberOfEvents() {
 
     }
 
+
+    /**
+     * Verifies that this peek batch of messages.
+     */
+    @Test
+    void peekBatchMessages() {
+        // Arrange
+        final int numberOfEvents = 2;
+
+        when(managementNode.peekBatch(numberOfEvents))
+            .thenReturn(Flux.fromArray(new ServiceBusReceivedMessage[]{message1, message2}));
+
+        // Act & Assert
+        StepVerifier.create(consumer.peekBatch(numberOfEvents))
+            .expectNextCount(numberOfEvents)
+            .verifyComplete();
+    }
+
+    /**
+     * Verifies that this peek batch of messages from a sequence Number.
+     */
+    @Test
+    void peekBatchWithSequenceNumberMessages() {
+        // Arrange
+        final int numberOfEvents = 2;
+        final int fromSequenceNumber = 10;
+
+        when(managementNode.peekBatch(numberOfEvents, fromSequenceNumber))
+            .thenReturn(Flux.fromArray(new ServiceBusReceivedMessage[]{message1, message2}));
+
+        // Act & Assert
+        StepVerifier.create(consumer.peekBatch(numberOfEvents, fromSequenceNumber))
```

Comment:
Instead of asserting the number of events, I would try to assert that the messages I expect came back. (ie. message 1 and message 2)